accepted/0003-plugin-manager.md

```diff
@@ -0,0 +1,114 @@
+knowingly used a experimental feature, etc.) or keep the plugin disabled. It
+would also be helpful to show an error code to the user to help them and the
+developer debug.
+
+### Installation
+To further ease the user with their plugin (once downloaded), installation could be handled in 3
+different ways.
+- A file explorer in the manager to install a compressed package that contains:
+  a manifest, locales, the actual plugin, and other needed components.
+- A file explorer that installs plugins by themselves - mostly for compatibilty
+  with previous plugins
+- A compressed package that is associated with OBS so when double clicked, it
+  can be automatically installed in the correct locations.
+
+If the user installs a compressed package, the package would be decompressed and
+the contents would be copied to a directory in the plugins folder.
+
+### Manifest
+As outlined in installed, a compressed package will contain multiple components
+including a manifest. The manifest will be responsible for outlining all
+important information about the plugin. An example of a barebones manifest is
+shown below:
+
+```json
+{
+    "name": "Plugin Name",
+    "author": "Author Name",
+    "description": "Description that would be displayed to user",
+    "locale": {
+        "en": "file path"
+    },
+    "plugin": {
+        "64": "file path",
+        "32": "file path"
+    },
+    "categories": ["PRODUCTIVITY", "CLASSIC", "THEMES"], //etc
+    "version": "^24.0.0"
+}
+```
+
+The purpose of the metadata in the manifest is for 3 main reasons: 1) Allowing
+for easier parsing of the compressed package, 2) allowing more flexibility for 
+plugin creators, and 3) reducing mental overhead for users organizing their
+plugins.
+
+Some notable entries on the manifest are categores and version. Version
+numbering allows the plugin creator to specifiy a specific version or range that
+the plugin can be loaded in. The categories affects the UX.
+
+Note: The options in the manifest are predefined. A plugin developer can not
+have random options that can be searched as outlined in UX.
+
+### UX
+To access the plugin manager, it would be under Settings -> Plugins or File ->
+Plugins. The plugin manager itself would have a searchbar at the top that allows
+searching through the metadata as provided in the manifest file and displays the
+results in the list below. Moreover, there would be a simple list view that
+outlines all the installed plugins in alphabetical order. The list would show
+the name, description, and categories of each plugin. All the way to the right,
+would be a button that outlines whether the plugin is disabled or currently
+active. Another feature would to allow the plugin to have its own settings menu
+which would display a button next to the disable button that opens its settings menu.
+
+When a user disables a plugin, it will add the plugin to a list to prevent
+loading during initilization. It will also attempt to unlink the plugin, but a
```

Comment:
Currently the plugins backend is not designed for dynamic unloading, and it may be dangerous to assume plugins will always fully clean up after themselves. It is most likely that the first version will require a restart to unload.